### ISSUE TITLE
doc_markdown gives false positive on intra-doc link paths with bracket syntax

### ISSUE THREAD

**Issue by TheIronBorn** [rust-lang/rust-clippy#7758]
Lint name: doc_markdown 

I tried this code:

```rust
/// [plain text][path::to::item]
```

I expected to see this happen: no warning emitted

Instead, this happened: 
```
warning: you should put `text][path::to::item` between ticks in the documentation
 --> src\lib.rs:1:12
  |
1 | /// [plain text][path::to::item]
  |            ^^^^^^^^^^^^^^^^^^^^
  |
  = note: `-W clippy::doc-markdown` implied by `-W clippy::pedantic`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#do
c_markdown
```

This doesn't happen with `()` syntax:
```rust
/// [plain text](path::to::item)
```

### Meta

**Rust version (`rustc -Vv`):**
```
rustc 1.57.0-nightly (c02371c44 2021-10-01)
binary: rustc
commit-hash: c02371c442f811878ab3a0f5a813402b6dfd45d2
commit-date: 2021-10-01
host: x86_64-pc-windows-msvc
release: 1.57.0-nightly
LLVM version: 13.0.0
clippy: 0.1.57 (c02371c4 2021-10-01)
```
